core/spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/function/HeaderTests.java

```diff
@@ -0,0 +1,159 @@
+	}
+
+	@Test
+	void checkWithEmptyPojo() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+			TestChannelBinderConfiguration.getCompleteConfiguration(EmptyConfiguration.class))
+			.web(WebApplicationType.NONE).run("--spring.jmx.enabled=false")) {
+
+			StreamBridge streamBridge = context.getBean(StreamBridge.class);
+			Message<EmptyPojo> message = MessageBuilder.withPayload(new EmptyPojo()).build();
+			streamBridge.send("emptyConfigurationDestination", message);
+
+			OutputDestination outputDestination = context.getBean(OutputDestination.class);
+			Message<byte[]> messageReceived = outputDestination.receive(1000, "emptyConfigurationDestination");
+			MessageHeaders headers = messageReceived.getHeaders();
+			assertThat(headers).isNotNull();
+			assertThat(headers.get(MessageUtils.TARGET_PROTOCOL)).isEqualTo("kafka");
+			assertThat(headers.get(MessageHeaders.CONTENT_TYPE)).isEqualTo("application/json");
+		}
+	}
+
+	@Test
+	void checkIfHeaderProvidedInData() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+			TestChannelBinderConfiguration.getCompleteConfiguration(EmptyConfiguration.class))
+			.web(WebApplicationType.NONE).run("--spring.jmx.enabled=false")) {
+			StreamBridge streamBridge = context.getBean(StreamBridge.class);
+			String jsonPayload = "{\"name\":\"Omer\"}";
+			streamBridge.send("myBinding-out-0",
+				MessageBuilder.withPayload(jsonPayload.getBytes())
+					.setHeader("anyHeader", "anyValue")
+					.build(),
+				MimeTypeUtils.APPLICATION_JSON);
+			OutputDestination output = context.getBean(OutputDestination.class);
+			Message<byte[]> result = output.receive(1000, "myBinding-out-0");
+			MessageHeaders headers = result.getHeaders();
+			assertThat(headers).isNotNull();
+			assertThat(headers.get(MessageUtils.TARGET_PROTOCOL)).isEqualTo("kafka");
+			assertThat(headers.get(MessageHeaders.CONTENT_TYPE)).isEqualTo("application/json");
+			assertThat(headers.get("anyHeader")).isEqualTo("anyValue");
+		}
+	}
+
+	@Test
+	void checkGenericMessageSent() {
+		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
+			TestChannelBinderConfiguration.getCompleteConfiguration(FunctionUpperCaseConfiguration.class))
+			.web(WebApplicationType.NONE)
+			.run("--spring.jmx.enabled=false",
+				"--spring.cloud.function.definition=uppercase")) {
+			String jsonPayload = "{\"surname\":\"Celik\"}";
+			InputDestination input = context.getBean(InputDestination.class);
+			input.send(new GenericMessage<>(jsonPayload.getBytes()), "uppercase-in-0");
+			OutputDestination output = context.getBean(OutputDestination.class);
+
+			Message<byte[]> result = output.receive(1000, "uppercase-out-0");
+			MessageHeaders headers = result.getHeaders();
+			assertThat(headers).isNotNull();
+			assertThat(headers.get(MessageUtils.TARGET_PROTOCOL)).isEqualTo("kafka");
+			assertThat(headers.get(MessageHeaders.CONTENT_TYPE)).isEqualTo("application/json");
+		}
+	}
+
+	@Test
```

Comment:
If the development I made is not implemented, only this test fails. The other tests complete successfully.